core/common/src/main/java/alluxio/heartbeat/HeartbeatThread.java

```diff
@@ -146,10 +156,15 @@ public void run() {
       while (!Thread.interrupted()) {
         // TODO(peis): Fix this. The current implementation consumes one thread even when ticking.
         mStatus = Status.WAITING;
+        mStartTickTime = CommonUtils.getCurrentMs();
         long limitTime = mTimer.tick();
         mStatus = Status.RUNNING;
-        LOG.debug("{} #{} will run limited in {}s", mThreadName, counter++, limitTime / 1000);
+        mStartHeartbeatTime = CommonUtils.getCurrentMs();
+        LOG.debug("{} #{} will run limited in {}s", mThreadName, mCounter.get(),
+            limitTime / Constants.SECOND_MS);
         mExecutor.heartbeat(limitTime);
+        mEndHeartbeatTime = CommonUtils.getCurrentMs();
+        updateState();
```

Comment:
every tick you are resetting to zero? looks like a bug?

Reply:
yeah, it looks need a structure to store the previous and current state